src-built-in/components/appCatalog2/src/app.jsx

```diff
@@ -214,27 +217,149 @@ export default class AppCatalog extends React.Component {
 					"friendlyName": "Dashboard of death"
 				}
 			],
-			activePage: "home"
+			filteredCards: [],
+			activePage: "home",
+			activeTags: [],
+			activeApp: null
 		};
 		this.bindCorrectContext();
 	}
 	bindCorrectContext() {
-		this.navigateToShowcase = this.navigateToShowcase.bind(this);
+		this.goHome = this.goHome.bind(this);
+		this.addTag = this.addTag.bind(this);
+		this.removeTag = this.removeTag.bind(this);
+		this.changeSearch = this.changeSearch.bind(this);
+		this.openAppShowcase = this.openAppShowcase.bind(this);
+		this.addApp = this.addApp.bind(this);
 	}
-	navigateToShowcase() {
+	addTag(tag) {
+		let tags = this.state.activeTags;
+		tags.push(tag);
 
+		let page = this.state.activePage;
+
+
+		if (tags.length === 1) {
+			page = "appShowcase";
+		}
+
+		this.setState({
+			activePage: page,
+			activeTags: tags
+		});
+	}
+	removeTag(tagName) {
+		let tags = this.state.activeTags.filter((tag) => {
+			return tag !== tagName;
+		});
+
+		let page = this.state.activePage;
+
+		if (tags.length === 0) {
+			page = "home";
+		}
+
+		this.setState({
+			activePage: page,
+			activeTags: tags
+		});
+	}
+	goHome() {
+		this.setState({
+			activePage: "home",
+			activeTags: [],
+			activeApp: null
+		});
+	}
+	changeSearch(search) {
+		if (search !== "") {
```

Comment:
I think `if(search !== "")` is not necessary here, `if(search)` should be sufficient